elixir/apps/api/lib/api/gateway/channel.ex

```diff
@@ -351,6 +351,15 @@ defmodule API.Gateway.Channel do
     end
   end
 
+  # Helper to directly send reject_access in integration tests
+  def handle_info(
+        {{:reject_access, gateway_id}, client_id, resource_id},
+        %{assigns: %{gateway: %{id: gateway_id}}} = socket
+      ) do
+    push(socket, "reject_access", %{client_id: client_id, resource_id: resource_id})
+    {:noreply, socket}
+  end
```

Comment:
This will simplify the call site a bit.

```suggestion
  def handle_info({:send_test_reject_access, client_id, resource_id}, socket) do
    push(socket, "reject_access", %{client_id: client_id, resource_id: resource_id})
  end
  
  def send_test_reject_access(resource_name) do
    [client_id | _] =
      Domain.Clients.Client
      |> Repo.all()
      |> Domain.Clients.preload_clients_presence()
      |> Enum.filter(&(&1.online?))
    resource =
      Domain.Resources.Resource
      |> Domain.Repo.all()
      |> Enum.filter(&(&1.name == resource_name))
      
    send(self(), {:send_test_reject_access, client_id, resource.id})
  end
```